### ISSUE TITLE
Priority for old session data in model binding

### ISSUE THREAD

**Issue by damms005** [spatie/laravel-html#153]
Is it possible that when a form element is generated, like a text input, if there is an item in the Session flash data matching the input name, that will take precedence over the model's value. So that there is a kind of priority for the binded model valus, pretty much like:

Session Flash Data (Old Input)
Model Attribute Data

So that it is easy tp re-populate forms when there is a validation error on the server?

**Comment by sebastiandedeyne** [comment]
You can always override the values by explicitly passing them.

In this example, `old('user_name')` won't be used.

```
{{ html()->model($user) }}

{{ html()->input('name')->value($user->name) }}
```

**Comment by damms005** [comment]
So what happens when `$user->user_name` is undefined but there is an `old('user_name')`?

**Comment by sebastiandedeyne** [comment]
Sorry, I think I misunderstood your issue. I thought you wanted to ignore whatever is in `old`.

Could you rephrase what you're trying to do? If possible with some code examples and expected output.

**Comment by damms005** [comment]
Description:

1. I have a generic `user_profile_form.blade.php` file that I use to add/edit users. It uses form model binding
2. When I load the view for existing user, everything works fine. Perfect!
3. Now, I load this same generic view for a new user, so for the model binding I just new-up a new `User` clas
4. User fills the `username` and `email` fields and submits form, but validation error causes redirect back

Expected:
The `username` filled previously should be filled

Current behavior:
The username field in the model-binded for form only countenances the `username` property (empty because new user), hence when validation redirection back occurs, the `username` field is blank.

Hence, my original suggestion that the logic should first check for Session Flash Data (Old Input), then Model Attribute Data (e.g. [LaravelCollective model form binding](https://laravelcollective.com/docs/6.0/html#form-model-binding))

**Comment by sebastiandedeyne** [comment]
Hmm looks like something's wrong then, because that *is* the default behaviour…

See https://github.com/spatie/laravel-html/blob/e138b02d964d15dfd33437b957df4755508d3835/src/Html.php#L583

It uses Laravel's `old` function under the hood. It tries to retrieve a session value, and if none are available, it retrieves it from the model.

Could you share the contents of `user_profile_form.blade.php`?

**Comment by damms005** [comment]
> Hmm looks like something's wrong then, because that is the default behaviour…
Then this issue can be closed then. I won't be able to share file content since I had since found a way around it (was an urgent feature do deploy at the time...couldn't wait). However, since this is the default behaviour, then there definitely was something wrong with me implementation at the time.

> It uses Laravel's old function under the hood. It tries to retrieve a session value, and if none are available, it retrieves it from the model.
I believe this should have a place in the readme doc.
